include/cantera/thermo/Phase.h

```diff
@@ -471,6 +471,23 @@ class Phase
     //! units = kg / kmol
     const vector_fp& molecularWeights() const;
 
+    //! Charge of species \c k.
+    //!     @param k index of species \c k
+    //!     @returns the charge of species \c k.
+    doublereal speciesCharge(size_t k) const;
+
+    //! Copy the vector of species charges into vector charges.
+    //!     @param charges Output vector of species charges (-)
+    void getSpeciesCharges(vector_fp& charges) const;
+
+    //! Copy the vector of species charges into array weights.
+    //!     @param weights Output array of species charges (-)
+    void getSpeciesCharges(doublereal* charges) const;
+
+    //! Return a const reference to the internal vector of molecular weights.
+    //! units = (-)
+    const vector_fp& speciesCharges() const;
```

Comment:
I know that this is being included as a parallel to the equivalent `molecularWeights()` method, but I don't think this method is necessary. In order to implement the other methods, you can use the `m_speciesCharge` array directly.

Reply:
Correct.